index.js

```diff
@@ -93,32 +93,71 @@ const discordClient = new Client({intents});
 
 discordClient.login(DISCORD_TOKEN);
 
-discordClient.on('ready', () =>{
-    console.log('discord client ready to go!')
-    discordClient.on("messageCreate", async message => {
-       console.log('message received');
+const onMessageHandler = async message => {
+    console.log('message received');
 
-        //if the message is from a bot, exit
-        if (message.author.bot) return;
+     //if the message is from a bot, exit
+     if (message.author.bot) return;
 
-        //if the message is a direct message, exit
-        if (message.type === 'dm') return;
+     //if the message is a direct message, exit
+     if (message.type === 'dm') return;
 
-        if (message.content.startsWith('!')) {
+     if (message.content.startsWith('!')) {
+          
+             let userMessage = message.content.substring(1)
+             let userInputRegex = /^\d{5}$/
              
-                let userMessage = message.content.substring(1)
-                let userInputRegex = /^\d{5}$/
+             if (userInputRegex.test(userMessage) ) {
+                 let weatherInfo = await getWeather(userMessage)
+                 let botMessage = `The current temperature in ${weatherInfo.city} is ${weatherInfo.temperature} degrees Fahrenheit.`;
+                 message.channel.send(botMessage);
+             }
+             else if (/\+\+/.test(userMessage)) {
+                console.log(`plusplus message ${userMessage}`);
+                // Discord user mention is translated to a user ID number
+
+                let username;
                 
-                if (userInputRegex.test(userMessage) ) {
-                    let weatherInfo = await getWeather(userMessage)
-                    let botMessage = `The current temperature in ${weatherInfo.city} is ${weatherInfo.temperature} degrees Fahrenheit.`;
-                    message.channel.send(botMessage);
+                if (/\+\+ <@\d+>/.test(userMessage)) {
+                    // Discord returned a user ID number, strip the leading <@
+                    const userIdPrefix = userMessage.slice(5);
+                    // and trailing >
+                    const userId = userIdPrefix.slice(0, userIdPrefix.length - 1);
+                    console.log(`userId ${userId}`);
+                    const user = await discordClient.users.fetch(userId, { cache: true });
+                    username = user.username;
```

Comment:
I'm pretty sure this'll work with the `let` above.

It is one of the rare cases where a semicolon is *actually* needed to say that the the parentheses don't represent a function call.

```suggestion
                    ;({ username } = user)
```